runtime/near-vm-logic/src/logic.rs

```diff
@@ -812,6 +813,93 @@ impl<'a> VMLogic<'a> {
         self.internal_write_register(register_id, value_hash)
     }
 
+    /// Compute multiexponentiation for (base, exp), serialized at bytes in LE notation.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + alt_bn128_g1_multiexp_base + alt_bn128_g1_multiexp_byte * num_bytes + alt_bn128_g1_multiexp_element * num_elements / (ln(min(1,num_elements))+1)`
+    pub fn alt_bn128_g1_multiexp(
+        &mut self,
+        value_len: u64,
+        value_ptr: u64,
+        register_id: u64,
+    ) -> Result<()> {
+        self.gas_counter.pay_base(alt_bn128_g1_multiexp_base)?;
+        let value_buf = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
+        self.gas_counter.pay_per_byte(alt_bn128_g1_multiexp_byte, value_buf.len() as u64)?;
+
+        let value = <Vec<(G1, Fr)>>::try_from_slice(&value_buf)
```

Comment:
We could create a special crate near-bn128 or something like this with binary serialized input and output and move all complex logic there. The same approach is used in openethereum for bls12 curves.

Reply:
I changed my mind about inclusion of `borsh`. Since `borsh` doesn't affect the gas cost for the runtime, we will be able to upgrade borsh as long as the serialization of the format doesn't change.

So let's keep `borsh`, but we probably can move serialization/deserialization of `bn` towards `nearcore`. This way it will stay explicit in the `nearcore` instead of being part of the `zeropool-bn` crate. We may need a wrapper for this, but it's should be fine.